jina/helper.py

```diff
@@ -972,6 +972,13 @@ def __get__(self, obj, cls):
         value = obj.__dict__[f'CACHED_{self.func.__name__}'] = self.func(obj)
         return value
 
+    def __delete__(self, obj):
+        cached_value = obj.__dict__.get(f'CACHED_{self.func.__name__}', None)
+        if cached_value is not None:
+            if hasattr(cached_value, 'close'):
```

Comment:
is there another way to check if the cached_value has to be closed.
Something with `isinstance()`

Reply:
What would be the instance? We have different classes that could be handlers here

Reply:
just brainstorming. If we can not make any assumptions about the handler class, then we have to keep it like this.

Reply:
not really a fan of this to be honest. We assume that close is an special method, which may not be true

Reply:
I think this is a code smell that we are caching something we should not cache